001-clarifications-eval.txt

```diff
@@ -0,0 +1,754 @@
+    6.4 Call Frame Information) operations are restricted to those that
+    do not require the compilation unit context to be specified.
+
+    The DWARF is ill-formed if all the address_size fields in the
+    headers of all the entries in the .debug_info, .debug_addr,
+    .debug_line, .debug_rnglists, .debug_rnglists.dwo, .debug_loclists,
+    and .debug_loclists.dwo sections corresponding to any given program
+    location do not match.
+    --------------------------------------------------------------------
+
+In Section 2.5.x (originally 2.5.1) General Operations, replace the
+first paragraph:
+
+    --------------------------------------------------------------------
+    Each general operation represents a postfix operation on a simple
+    stack machine. Each element of the stack has a type and a value, and
+    can represent a value of any supported base type of the target
+    machine. Instead of a base type, elements can have a generic type,
+    which is an integral type that has the size of an address on the
+    target machine and unspecified signedness. The value on the top of
+    the stack after “executing” the DWARF expression is taken to be the
+    result (the address of the object, the value of the array bound, the
+    length of a dynamic string, the desired value itself, and so on).
+    --------------------------------------------------------------------
+
+... with the following:
+
+    --------------------------------------------------------------------
+    Each general operation represents a postfix operation on a simple
+    stack machine.
+
+    A value on the stack has a type and a literal value. It can
+    represent a literal value of any supported base type of the target
+    architecture. The base type specifies the size, encoding, and
+    endianity of the literal value.
+
+    There is a distinguished base type termed the generic type, which is
+    an integral type that has the size of an address in the target
+    architecture default address space, a target architecture defined
+    endianity, and unspecified signedness.
+
+    [non-normative] The generic type is the same as the unspecified type
+    used for stack operations defined in DWARF Version 4 and before.
+
+    An integral type is a base type that has an encoding of
+    DW_ATE_signed, DW_ATE_signed_char, DW_ATE_unsigned,
+    DW_ATE_unsigned_char, DW_ATE_boolean, or any target architecture
+    defined integral encoding in the inclusive range DW_ATE_lo_user to
+    DW_ATE_hi_user.
+    --------------------------------------------------------------------
+
+    > [For further discussion...]
+    > It may be desirable to add an implicit pointer base type encoding.
+    > It would be used for the type of the value that is produced when
+    > the DW_OP_deref* operation retrieves the full contents of an
+    > implicit pointer location storage created by the
+    > DW_OP_implicit_pointer operation. The literal value would record
```

Comment:
DW_OP_implicit_pointer defines a location description.  One cannot apply DW_OP_deref* on it as the latter expects an address-sized value on the stack and interprets it as memory address.  Such considerations belong into section 4 or later.

Reply:
In 004-locations-on-stack.txt the DW_OP_deref* operations are defined to work on location descriptions. So I think this note should be moved to 004-locations-on-stack.txt .